experimental/FTheoryTools/src/AbstractFTheoryModels/attributes.jl

```diff
@@ -1195,33 +1186,222 @@ function chern_class_c2(m::AbstractFTheoryModel; check::Bool = true)
     return get_attribute(m, :chern_class_c2)
   end
 
-  # Trigger potential short-cut computation of cohomology ring
-  cohomology_ring(ambient_space(m); check)
+  # Compute elements of the cohomology ring corresponding to the torus invariant prime divisors
+  c_ds = [g.f for g in gens(cohomology_ring(ambient_space(m); check = false))]
+  R = parent(c_ds[1])
+  projection_map = hom(R, cohomology_ring(ambient_space(m); check = false), gens(R))
+  len_c_ds = length(c_ds)
 
   # Compute the cohomology class corresponding to the hypersurface equation
-  if m isa WeierstrassModel
-    cl = toric_divisor_class(ambient_space(m), degree(weierstrass_polynomial(m)))
+  cy = cohomology_class(toric_divisor_class(ambient_space(m), degree(hypersurface_equation(m))))
+
+  # Identify index combination that may be discarded by virtue of the Stanley-Reisner ideal
+  mnf = Oscar._minimal_nonfaces(ambient_space(m))
+  indices = [findall(x -> x == 1, Vector{Int}(mnf[i, :])) for i in 1:Polymake.nrows(mnf)]
```

Comment:
More efficient:
```suggestion
  indices = [Polymake.SparseArrays.nonzeroinds(mnf[i, :]) for i in 1:Polymake.nrows(mnf)]
```

Old code:
```
julia> @benchmark [findall(x -> x == 1, Vector{Int}(mnf[i, :])) for i in 1:Polymake.nrows(mnf)]
BenchmarkTools.Trial: 1265 samples with 1 evaluation.
 Range (min … max):  2.251 ms …   1.898 s  ┊ GC (min … max): 0.00% … 15.24%
 Time  (median):     2.441 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.948 ms ± 53.295 ms  ┊ GC (mean ± σ):  5.79% ±  0.43%

                 ▃▁▄▇▆▅▆█▅█▅▅▄▄▃▃▃▁
  ▂▂▁▂▂▁▂▂▄▄▆▄▆█████████████████████▇▆▄▄▄▆▃▄▄▄▃▂▂▂▂▃▂▂▁▂▁▂▁▁ ▄
  2.25 ms        Histogram: frequency by time         2.7 ms <

 Memory estimate: 398.42 KiB, allocs estimate: 14711.
```

Suggested alternative:
```
julia> @benchmark [Polymake.SparseArrays.nonzeroinds(mnf[i, :]) for i in 1:Polymake.nrows(mnf)]
BenchmarkTools.Trial: 1608 samples with 1 evaluation.
 Range (min … max):  1.677 ms …   2.773 s  ┊ GC (min … max): 0.00% … 14.50%
 Time  (median):     1.861 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.593 ms ± 69.108 ms  ┊ GC (mean ± σ):  6.96% ±  0.36%

               ▁▃▂▃▇▅▆▇▆▆█▇▅▃▃▂ ▂
  ▂▂▂▃▂▃▃▄▄▅▆▇███████████████████▇▅▆▆▅▄▄▄▃▄▄▃▃▃▃▃▁▂▂▃▁▃▂▁▂▂▂ ▅
  1.68 ms        Histogram: frequency by time        2.15 ms <

 Memory estimate: 275.61 KiB, allocs estimate: 15056.
```


Annoyinglh in both cases the majority of the overhead comes from computing `mnf[i,:]`:
```
julia> @benchmark [mnf[i, :] for i in 1:Polymake.nrows(mnf)]
BenchmarkTools.Trial: 2018 samples with 1 evaluation.
 Range (min … max):  1.541 ms …   3.434 s  ┊ GC (min … max): 0.00% … 14.05%
 Time  (median):     1.744 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.465 ms ± 76.413 ms  ┊ GC (mean ± σ):  6.90% ±  0.31%

            ▂▂▃▆▇█▇▅▆▆▆▆▅▅▆▄▃▁▁   ▁
  ▂▂▃▃▃▄▄▅▆████████████████████▇▆▇█▅▆▆▅▄▄▄▄▄▃▃▃▃▃▂▂▂▂▃▂▂▂▂▁▂ ▅
  1.54 ms        Histogram: frequency by time        2.12 ms <

 Memory estimate: 219.48 KiB, allocs estimate: 13343.

```

Maybe @benlorenz knows a way to get this data with less overhead?

Reply:
Thank you. I will work this in.